compiler/lib/generate.ml

```diff
@@ -950,7 +950,14 @@ let parallel_renaming params args continuation queue =
 
 let apply_fun_raw ctx f params exact cps =
   let n = List.length params in
-  let apply_directly = J.call f params J.N in
+  let apply_directly =
+    (* Make sure we are performing a regular call, not a (slower)
+       method call *)
```

Comment:
@vouillon, can you provide some reference mentioning this optimization. I did some test and it seems to show the opposite.

Reply:
I don't have any reference, but without this, the improved analysis resulted in slower code.
I just tried is compiling `ocamlc.byte` (`dune exec -- js_of_ocaml --opt 3 `which ocamlc.byte` -o /tmp/ocamlc.js`).
And then running it on some ml source files:
```
time node /tmp/ocamlc.js -c ./benchmarks/sources/ml/*.ml ./benchmarks/sources/ml/*.ml ./benchmarks/sources/ml/*.ml
```
This optimization makes close to a 8% performance improvement on my machine.

Reply:
What ocaml version do you use ? it seems that ocamlc no longer works when compiled to js due to https://github.com/ocaml/ocaml/pull/11997 (since OCaml 5.1). The reason is that Reloc_literal can now contain floats that jsoo doesn't want to marshal.

Reply:
I'm using `python3 -m pyperf system tune` to reduce the noise.